packages/shared/client/ReplayClient.ts

```diff
@@ -878,6 +887,36 @@ export class ReplayClient implements ReplayClientInterface {
     }
   }
 
+  async waitForLoadedRegions(focusRange: TimeStampedPointRange | PointRange | null): Promise<void> {
+    return new Promise(resolve => {
+      const checkLoaded = () => {
+        const loadedRegions = this.loadedRegions;
+        let isLoaded = false;
+        if (loadedRegions !== null) {
+          if (focusRange !== null) {
+            const pointRange = toPointRange(focusRange);
+
+            isLoaded = isRangeInRegions(pointRange.begin, pointRange.end, loadedRegions.loaded);
+          } else {
+            isLoaded = loadedRegions.loading.length === 0;
```

Comment:
I'm not sure that it matters, because I suspect it's impossible (at least, I hope so), but the case where there are no regions being loaded seems like it would be an error.

Reply:
Good thought. I'll add a guard for it.